woocommerce/woocommerce-gutenberg-products-block
# Abort inflight batch requests if another one comes with same paths

This PR is based on `try/decouple-customer-data-context` so not all changes to trunk are visible here.

POC WIP



Fixes #




#### Accessibility




#### Other Checks


### Screenshots



### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:
1. Ensure you have two or three shipping methods set up.
2. Add an item to your cart and go to the Cart block.
3. Open the network tab of dev tools.
4. Change shipping rates quickly (but slowly enough that a batch request is sent between each one)
5. Observe the dev tools cancelling the batch request when a new one is started.
6. Try to update the quantity of a cart item then very quickly change shipping method while the quantity change batch is in flight.
7. Ensure the first request is not cancelled.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1. 
2.
3.



### Performance Impact



### Changelog

> Add suggested changelog entry here.
